ygot/struct_validation_map_test.go

```diff
@@ -872,3 +872,315 @@ func TestMergeStructJSON(t *testing.T) {
+		},
+	}, {
+		name: "struct pointer with single field",
+		inSrc: &copyTest{
+			StringField: String("lagunitas-aunt-sally"),
+			StructPointer: &copyTest{
+				StringField: String("deschutes-pinedrops"),
+			},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringField: String("lagunitas-aunt-sally"),
+			StructPointer: &copyTest{
+				StringField: String("deschutes-pinedrops"),
+			},
+		},
+	}, {
+		name: "struct pointer with multiple fields",
+		inSrc: &copyTest{
+			StringField: String("allagash-brett"),
+			Uint32Field: Uint32(84),
+			StructPointer: &copyTest{
+				StringField: String("brooklyn-summer-ale"),
+				Uint32Field: Uint32(128),
+			},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringField: String("allagash-brett"),
+			Uint32Field: Uint32(84),
+			StructPointer: &copyTest{
+				StringField: String("brooklyn-summer-ale"),
+				Uint32Field: Uint32(128),
+			},
+		},
+	}, {
+		name: "enum value",
+		inSrc: &copyTest{
+			EnumValue: EnumTypeValue,
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			EnumValue: EnumTypeValue,
+		},
+	}, {
+		name: "union field",
+		inSrc: &copyTest{
+			UnionField: &copyUnionS{"new-belgium-fat-tire"},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			UnionField: &copyUnionS{"new-belgium-fat-tire"},
+		},
+	}, {
+		name: "string slice",
+		inSrc: &copyTest{
+			StringSlice: []string{"sierra-nevada-pale-ale", "stone-ipa"},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringSlice: []string{"sierra-nevada-pale-ale", "stone-ipa"},
+		},
+	}, {
+		name: "string map",
+		inSrc: &copyTest{
+			StringMap: map[string]*copyTest{
+				"ballast-point": &copyTest{StringField: String("sculpin")},
+				"upslope":       &copyTest{StringSlice: []string{"amber-ale", "brown"}},
+			},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StringMap: map[string]*copyTest{
+				"ballast-point": &copyTest{StringField: String("sculpin")},
+				"upslope":       &copyTest{StringSlice: []string{"amber-ale", "brown"}},
+			},
+		},
+	}, {
+		name: "struct map",
+		inSrc: &copyTest{
+			StructMap: map[copyMapKey]*copyTest{
+				copyMapKey{"saint-arnold"}: &copyTest{StringField: String("fancy-lawnmower")},
+				copyMapKey{"green-flash"}:  &copyTest{StringField: String("hop-head-red")},
+			},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StructMap: map[copyMapKey]*copyTest{
+				copyMapKey{"saint-arnold"}: &copyTest{StringField: String("fancy-lawnmower")},
+				copyMapKey{"green-flash"}:  &copyTest{StringField: String("hop-head-red")},
+			},
+		},
+	}, {
+		name: "struct slice",
+		inSrc: &copyTest{
+			StructSlice: []*copyTest{{
+				StringField: String("russian-river-pliny-the-elder"),
+			}, {
+				StringField: String("lagunitas-brown-shugga"),
+			}},
+		},
+		inDst: &copyTest{},
+		wantDst: &copyTest{
+			StructSlice: []*copyTest{{
+				StringField: String("russian-river-pliny-the-elder"),
+			}, {
+				StringField: String("lagunitas-brown-shugga"),
+			}},
+		},
```

Comment:
Case where struct has list and the non-list members are equal but list elements have partial overlap?

Reply:
Added test that return errors to verify.